pkg/specgen/generate/container.go

```diff
@@ -481,7 +489,68 @@ func ConfigToSpec(rt *libpod.Runtime, specg *specgen.SpecGenerator, contaierID s
 		}
 	}
 	specg.OverlayVolumes = overlay
-	specg.Mounts = conf.Spec.Mounts
+	name, mounts := c.SortUserVolumes(c.Spec())
+	tempInspect, err := c.GetMounts(name, conf.ImageVolumes, mounts)
+	if err != nil {
+		return nil, nil, err
+	}
+	mountsToPass := []string{}
+	for _, val := range tempInspect {
+		mountsToPass = append(mountsToPass, fmt.Sprintf("type=%v,src=%v,target=%v,%v", val.Type, val.Source, val.Destination, strings.Join(val.Options, ",")))
+	}
+	m, _, _, err := specgenutil.Mounts(mountsToPass)
+
+	// a lot of options get added later and are hard to translate.
+	// weed those out here
+	for err != nil && strings.Contains(err.Error(), "invalid mount option") {
```

Comment:
Can we use proper error comparison here with something like errors.Is().
This will break as soon as someone changes the error message.